main.go

```diff
@@ -84,6 +87,81 @@ func buildUI() fyne.CanvasObject {
 		}),
 		widget.NewToolbarAction(theme.MailForwardIcon(), func() {
 			code := fmt.Sprintf("%#v", overlay)
+
+			layoutReplace := regexp.MustCompile(`(layout.[a-zA-Z]+)`)
+			code = layoutReplace.ReplaceAllString(code, "${1}Layout") // ToDo: should remove this line once the right layout is picked
+
+			funcReplace := regexp.MustCompile(`\(func\(\)\)\([a-zA-Z0-9]*\)`)
+			code = funcReplace.ReplaceAllString(code, "func(){fmt.Println(\"Hello there\")}")
+
+			areSimplePropsPresent := regexp.MustCompile(`[{ ]+[a-z][a-zA-Z]*:`)
+			simpleProps1 := regexp.MustCompile(`([{ ]+)([a-z][a-zA-Z]*:[a-zA-Z0-9]*,)`)
+			simpleProps2 := regexp.MustCompile(`([{ ]+)([a-z][a-zA-Z]*:[a-zA-Z0-9]*)([},]+)`)
+			simpleProps3 := regexp.MustCompile(`([{ ]+)([a-z][a-zA-Z]*:[ ]*[a-z.A-Z]*{[a-zA-Z:0-9.,\- ]+}[,]*)`)
+			simpleProps4 := regexp.MustCompile(`([{ ]+)([a-z][a-zA-Z]*:[ ]*[a-zA-Z0-9\.\*]*\([0-9a-zA-Z]+\)[,]*)`)
+			simpleProps5 := regexp.MustCompile(`([{ ]+)([a-z][a-zA-Z]*:[ ]*\([a-zA-Z0-9\.\*]*\)\([0-9a-zA-Z]+\)[,]*)`)
+			simpleProps6 := regexp.MustCompile(`([{ ]+)([a-z][a-zA-Z]*:[ ]*\[\][a-z]+\([a-zA-Z0-9, "]+\)[,]*)`)
+			simpleProps7 := regexp.MustCompile(`([{ ]+)([a-z][a-zA-Z]*:[ ]*\(func\([a-zA-Z.]+\)\)\([0-9a-zA-Z]+\)[,]*)`)
+			simpleProps8 := regexp.MustCompile(`([{ ]+)([a-z][a-zA-Z]*:[ ]*func\(\){fmt.Println\("Hello there"\)}[,]*)`)
+			simpleProps9 := regexp.MustCompile(`([{ ]+)([a-z][a-zA-Z]*:[ ]*map\[[a-z]+\][a-z0-9]+[\({]+[a-zA-Z0-9:, "]+[\)}]+[,]*)`)
+			simpleProps10 := regexp.MustCompile(`([{ ]+)(shortcut:[ ]*fyne.ShortcutHandler{.*?Copy.*?Cut.*?Paste.*?SelectAll.*?}}[,]*)`)
+
+			for areSimplePropsPresent.MatchString(code) {
+				code = simpleProps1.ReplaceAllString(code, "$1")
+				code = simpleProps2.ReplaceAllString(code, "$1$3")
+				code = simpleProps3.ReplaceAllString(code, "$1")
+				code = simpleProps4.ReplaceAllString(code, "$1")
+				code = simpleProps5.ReplaceAllString(code, "$1")
+				code = simpleProps6.ReplaceAllString(code, "$1")
+				code = simpleProps7.ReplaceAllString(code, "$1")
+				code = simpleProps8.ReplaceAllString(code, "$1")
+				code = simpleProps9.ReplaceAllString(code, "$1")
+				code = simpleProps10.ReplaceAllString(code, "$1")
+
+			}
+			fmt.Println(areSimplePropsPresent.MatchString(code))
+
+			iconReplace := regexp.MustCompile(`\(\*theme.ThemedResource\)\((0x[a-zA-Z0-9]+)\)`)
+			fmt.Println("=================")
+
+			cancelNewAddr, _ := strconv.ParseUint(fmt.Sprintf("%p", theme.DefaultTheme().Icon("cancel")), 0, 64)
+
+			for _, v := range iconReplace.FindAllStringSubmatch(code, -1) {
+				iconAddr, _ := strconv.ParseUint(v[1], 0, 64)
+				strAddr := "0x" + strconv.FormatUint(cancelAddr+iconAddr-cancelNewAddr, 16)
```

Comment:
I't not chear what this line does. Perhaps "cancelNewAddr" is not a clear variable name?